src/main/java/jeresources/compatibility/minecraft/MobCompat.java

```diff
@@ -0,0 +1,87 @@
+import net.minecraft.entity.Entity;
+import net.minecraft.entity.EntityType;
+import net.minecraft.entity.LivingEntity;
+import net.minecraft.entity.MobEntity;
+import net.minecraft.entity.boss.dragon.EnderDragonEntity;
+import net.minecraft.entity.monster.BlazeEntity;
+import net.minecraft.entity.monster.GuardianEntity;
+import net.minecraft.entity.monster.MagmaCubeEntity;
+import net.minecraft.entity.monster.MonsterEntity;
+import net.minecraft.entity.monster.PatrollerEntity;
+import net.minecraft.entity.monster.PhantomEntity;
+import net.minecraft.entity.monster.PillagerEntity;
+import net.minecraft.entity.monster.SlimeEntity;
+import net.minecraft.entity.monster.VexEntity;
+import net.minecraft.entity.monster.WitchEntity;
+import net.minecraft.entity.passive.AnimalEntity;
+import net.minecraft.entity.passive.BatEntity;
+import net.minecraft.entity.passive.DolphinEntity;
+import net.minecraft.entity.passive.WaterMobEntity;
+import net.minecraft.world.World;
+import net.minecraftforge.registries.ForgeRegistries;
+
+public class MobCompat {
+    private static World world = MinecraftCompat.getWorld();
+    private static final HashMap<Class, Integer[]> MOB_XP = new HashMap<>();
+    private static final HashMap<Class, LightLevel> LIGHT_LEVEL = new HashMap<>();
+
+    public static void setLightLevel(MobEntry entry) {
+        Class entityClass = entry.getEntity().getClass();
+        entry.setLightLevel(LIGHT_LEVEL.get(entityClass));
+    }
+
+    public static void setExperience(MobEntry entry) {
+        Class entityClass = entry.getEntity().getClass();
+        System.out.println("The Entity Class: "+entityClass+" | The Mininum Exp: "+MOB_XP.get(entityClass)[0]);
+        entry.setMinExp(MOB_XP.get(entityClass)[0]);
+        entry.setMaxExp(MOB_XP.get(entityClass)[1]);
+    }
+
+    static {
+        for (EntityType entityType : ForgeRegistries.ENTITIES) {
+            Entity entity = entityType.create(world);
+
+            if (entity instanceof MobEntity) {
+                Class entityClass = entity.getClass();
+                Integer[] exp = new Integer[2];
+
+                if(entity instanceof AnimalEntity || entity instanceof WaterMobEntity) {
+                    exp[0] = 1;
+                    exp[1] = 3;
+                }
+                else if(entity instanceof EnderDragonEntity) {
+                    exp[0] = 500;
+                    exp[1] = 12000;
+                }
+                else if(entity instanceof SlimeEntity || entity instanceof MagmaCubeEntity) {
+                    exp[0] = 1;
+                    exp[1] = 4;
+                }
+                else {
+                    exp[0] = exp[1] = ((MobEntity)entity).experienceValue;
+                }
+
+                MOB_XP.put(entityClass, exp);
+
+                if (entity instanceof VexEntity || entity instanceof GuardianEntity || (entity instanceof PatrollerEntity && !(entity instanceof WitchEntity))) //all the entities that cannot spawn naturally

```

Comment:
Would so make this big if statement a function call. Some thing like
 ```private boolean cannotSpawnNaturally(Class entityClass)``` 
this will also improve readability